### PR TITLE
Ct 298: add requester type

### PR DESCRIPTION
**Pull request by EdwardAndress**
ct-298

- It's useful for those dealing with a case to know something about who created it
- Added requester_type as an enum to Case
- updated factory

Also, the rake cases:demo_entries was attempting to use an invalid factory
- fixed by pointing at :case, rather than :cases

### CODE REVIEW

File: app/views/cases/new.html.slim [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-
 = GovukElementsErrorsHelper.error_summary @case,
   "#{pluralize(@case.errors.count, t('common.error'))} #{ t('common.summary_error')}", ""
 
+= render partial: 'layouts/header', locals: {page_heading: t('.new_case')}
+
 = form_for @case do |f|
 
+  = f.radio_button_fieldset :requester_type, choices: [:academic_business_charity, :journalist, :member_of_the_public, :offender, :solicitor, :staff_judiciary, :what_do_they_know], inline: false
+
```
Comment (aliuk2012) [inline app/views/cases/new.html.slim]: You shouldn't need `inline: false`

File: db/structure.sql [hunk 2]
```diff
@@ -370,6 +387,26 @@ ALTER TABLE ONLY cases
...
+('20161017120533'),
+('20161031133532'),
+('20161103104520'),
+('20161114143107'),
+('20161116150744'),
+('20161116153411'),
+('20161125115930'),
+('20161130164018'),
+('20161209220224'),
+('20170111115617');
```
Comment (aliuk2012) [inline db/structure.sql]: multiline new lines, the same problem we had with #66 commit this morning

**Comment by aliuk2012** [comment]
Any particular reason why theres loads of other changes in db/structure.sql in terms of 'cases' and categories etc

File: db/structure.sql (hunk 2 above)
Comment (EdwardAndress) [inline db/structure.sql]: argh!  that sneaked in, sorry

File: app/views/cases/new.html.slim (hunk 1 above)
Comment (EdwardAndress) [inline app/views/cases/new.html.slim]: OK I will remove it

File: db/structure.sql (hunk 2 above)
Comment (EdwardAndress) [inline db/structure.sql]: Actually... shouldn't this be in?  Maybe the problem is that some previous changes were not checked in.

**Comment by EdwardAndress** [comment]
@aliuk2012 regarding db/structure.sql, since the file is generated automatically I am not sure why there may be unexpected changes... @misaka ?

**Comment by misaka** [comment]
@EdwardAndress @aliuk2012 Sounds like the changes to `structure.sql` are likely related to the version. Once we get #73 merged let's fix up the `structure.sql` here (dump it again using pg_dump 9.5.5) and rebase the changes and see how far that gets us.

**Comment by EdwardAndress** [comment]
I used the web-based method for resolving a merge conflict, as an experiment, without realising it would generate a merge commit.

**Comment by EdwardAndress** [comment]
Otherwise, are we all happy with this now?

File: db/migrate/20170111115617_add_requester_type_to_case.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    create_enum :requester_type,
+      'academic_business_charity',
+      'journalist',
+      'member_of_the_public',
+      'offender',
+      'solicitor',
+      'staff_judiciary',
+      'what_do_they_know'
+
+    add_column :cases, :requester_type, :requester_type
```
Comment (misaka) [inline db/migrate/20170111115617_add_requester_type_to_case.rb]: What are the chances we'll need to create a filter in the front-end based on this column? If so, we may want to consider putting an index on here, depending on how confident we feel that we'll remember to index it later when we need to add the filter.

File: spec/features/02_create/case_create_spec.rb [hunk 4]
```diff
@@ -23,6 +24,8 @@
   scenario 'succeeds using valid inputs' do
     expect(current_path).to eq new_case_path
     expect(page).to have_content('New case')
+
+    choose user_input.requester_type
```
Comment (misaka) [inline spec/features/02_create/case_create_spec.rb]: I don't see where we're testing that this was saved in the case below, are we?

File: db/migrate/20170111115617_add_requester_type_to_case.rb (hunk 3 above)
Comment (EdwardAndress) [inline db/migrate/20170111115617_add_requester_type_to_case.rb]: There may be other places where indexing is also missing.  assignment_type for example

File: spec/features/02_create/case_create_spec.rb (hunk 4 above)
Comment (EdwardAndress) [inline spec/features/02_create/case_create_spec.rb]: I wonder whether that was lost when I tried to resolve a messy MC.  Anyway, I will add it in now.

**Pull request merged**